### PR TITLE
fix for issue breaking `make` on mac: Update macOS deployment target to 12.0 and bottle version to Monterey

### PR DESCRIPTION
**Pull request by iurimatias**
### What does the PR do

The openssl bigsur bottle has be dropped, this PR updates the Makefile so the minimum version is now Monterey

### CODE REVIEW

**Review by igor-sirotin — APPROVED** [review]
There's also a bug in `fetch-brew-bottle.sh`, so we didn't see a proper message about bottle missing in Homerbrew.
It can be fixed like this:

```diff
--- a/scripts/fetch-brew-bottle.sh
+++ b/scripts/fetch-brew-bottle.sh
@@ -53,8 +53,8 @@ BOTTLE_JSON=$(get_bottle_json "${BOTTLE_NAME}" "${BOTTLE_FILTER}")
 BOTTLE_URL=$(echo "${BOTTLE_JSON}" | jq -r .url)
 BOTTLE_SHA=$(echo "${BOTTLE_JSON}" | jq -r .sha256)
 
-if [[ -z "${BOTTLE_URL}" ]] || [[ -z "${BOTTLE_SHA}" ]]; then
-    echo "Failed to identify bottle URL or SHA256!" >&2
+if [[ "${BOTTLE_URL}" == "null" ]] || [[ "${BOTTLE_SHA}" == "null" ]]; then
+    echo "Failed to identify bottle URL or SHA256! Bottle dropped from Homebrew?" >&2
     exit 1
 fi
```

So we would see a friendly message in such case:
```
Fetching: openssl@1.1 bottle arch arm64 'arm64_big_sur'
./scripts/fetch-brew-bottle.sh openssl@1.1 'arm64_big_sur' 
openssl@1.1 - Fetching GH Pkgs Token
openssl@1.1 - Finding bottle URL
openssl@1.1 - Selecting: arm64_big_sur
openssl@1.1 - Found bottle json data. URL: null, SHA: null
Failed to identify bottle URL or SHA256! Bottle dropped from Homebrew?
Failed to download bottle. Check your authentication token and network connection.
```

File: scripts/fetch-brew-bottle.sh [hunk 1]
```diff
@@ -53,8 +54,8 @@ BOTTLE_JSON=$(get_bottle_json "${BOTTLE_NAME}" "${BOTTLE_FILTER}")
 BOTTLE_URL=$(echo "${BOTTLE_JSON}" | jq -r .url)
 BOTTLE_SHA=$(echo "${BOTTLE_JSON}" | jq -r .sha256)
 
-if [[ -z "${BOTTLE_URL}" ]] || [[ -z "${BOTTLE_SHA}" ]]; then
-    echo "Failed to identify bottle URL or SHA256!" >&2
+if [[ "${BOTTLE_URL}" == "null" ]] || [[ "${BOTTLE_SHA}" == "null" ]]; then
+    echo "Failed to identify bottle URL or SHA256! Bottle dropped from Homebrew?" >&2
```
Comment (iurimatias) [inline scripts/fetch-brew-bottle.sh:58]: @igor-sirotin updated

**Review by caybro — COMMENTED** [review]
In storybook, we have:
```
if (APPLE)
    set(MACOS_VERSION_MIN_FLAGS -mmacosx-version-min=10.14)
endif()
```

File: scripts/fetch-brew-bottle.sh (hunk 1 above)
Comment (igor-sirotin) [reply scripts/fetch-brew-bottle.sh:58]: thanks 👍

File: scripts/fetch-brew-bottle.sh [hunk 2]
```diff
@@ -10,6 +10,7 @@ function get_gh_pkgs_token() {
 }
 
 function get_bottle_json() {
+    # echo "Getting bottle JSON for $1 $2"
```
Comment (igor-sirotin) [inline scripts/fetch-brew-bottle.sh:13]: leftover?

**Review by igor-sirotin — APPROVED** [review]

**Review by caybro — CHANGES_REQUESTED** [review]
Similarly, in config.nims:
```
  # set the minimum supported macOS version to 11.0
  switch("passC", "-mmacosx-version-min=11.0")
```

File: scripts/fetch-brew-bottle.sh (hunk 2 above)
Comment (iurimatias) [reply scripts/fetch-brew-bottle.sh:13]: yes, will remove when updating storybook

**Review by dlipicar — APPROVED** [review]

**Review by caybro — COMMENTED** [review]
Still the storybook cmake file left 🙂

**Comment by iurimatias** [comment]
> In storybook, we have:
> 
> ```
> if (APPLE)
>     set(MACOS_VERSION_MIN_FLAGS -mmacosx-version-min=10.14)
> endif()
> ```

wondering if it's worth it / could introduce issues, since storybook is a separate thing

**Comment by igor-sirotin** [comment]
> Similarly, in config.nims:

I don't like that we store this in 2 places, but for now I'd just update it in both places and leave a comment not to forget next time 😄 

@iurimatias it's here:
https://github.com/status-im/status-desktop/blob/e698fd3e34b97c369bcd661d88709085624138f0/config.nims#L26-L27

**Review by caybro — APPROVED** [review]
👍

**Comment by caybro** [comment]
> > In storybook, we have:
> > 
> > ```
> > if (APPLE)
> >     set(MACOS_VERSION_MIN_FLAGS -mmacosx-version-min=10.14)
> > endif()
> > ```
> 
> wondering if it's worth it / could introduce issues, since storybook is a separate thing

It probably doesn't matter much but better keep them in sync

File: storybook/CMakeLists.txt [hunk 3]
```diff
@@ -15,7 +15,7 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
 if (APPLE)
-    set(MACOS_VERSION_MIN_FLAGS -mmacosx-version-min=10.14)
+    set(MACOS_VERSION_MIN_FLAGS -mmacosx-version-min=11.0)
```
Comment (igor-sirotin) [inline storybook/CMakeLists.txt:18]: Wait a second, why not `12.0`? 😄

**Pull request merged**